addons/web/static/tests/views/form_tests.js

```diff
@@ -7016,6 +7016,52 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('autoresize of text fields is done on notebook page show', async function (assert) {
+        assert.expect(5);
+
+        this.data.partner.fields.text_field = { string: 'Text field', type: 'text' };
+        this.data.partner.fields.text_field.default = "some\n\nmulti\n\nline\n\ntext\n";
+        this.data.partner.records[0].text_field = "a\nb\nc\nd\ne\nf";
+
+        var form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form string="Partners">' +
+                    '<sheet>' +
+                        '<notebook>' +
+                            '<page string="Choucroute">' +
+                                '<field name="foo"/>' +
+                            '</page>' +
+                            '<page string="Cassoulet">' +
+                                '<field name="text_field"/>' +
+                            '</page>' +
+                        '</notebook>' +
+                    '</sheet>' +
+                '</form>',
+            res_id: 1,
+        });
+
+        await testUtils.form.clickEdit(form);
+        assert.hasClass(form.$('.o_notebook .nav .nav-link:first()'), 'active');
+
+        await testUtils.dom.click(form.$('.o_notebook .nav .nav-link:nth(1)'));
+        assert.hasClass(form.$('.o_notebook .nav .nav-link:nth(1)'), 'active');
+
+        var height = form.$('.o_field_widget[name=text_field]').height();
+        assert.ok(height > 80, "textarea should have an height of at least 80px");
+
+        assert.containsN(form, 'textarea', 2,
+            "should have two textarea one for field and one fixed textarea");
+        // 2-2 px top-bottom padding in o_input element i.e. 0th textarea, check scrollHeight of
+        // textarea is equal to fixedTextarea created by dom.autoresize function
+        assert.strictEqual(form.$('textarea:eq(0)')[0].scrollHeight + 'px',
+                form.$('textarea:eq(1)')[0].scrollHeight + 4 + 'px',
+                "autoresized textarea should have height of fixed textarea + padding (0 line)");
```

Comment:
the last two asserts should be removed I think (there rely on implementation details). I think that testing that the height is at least 80 is enough. However, i'd add another (empty) textarea somewhere else in the view and assert that it's height is 50, to ensure that the > 80 assert is relevant.

Reply:
@aab-odoo Done 👍